pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -175,14 +205,20 @@ func (s *spdxParser) getFileElement(elementID string) []*model.PkgInputSpec {
 
 func (s *spdxParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
 	logger := logging.FromContext(ctx)
-
 	preds := &assembler.IngestPredicates{}
 
-	toplevel := s.getPackageElement("DOCUMENT")
+	topLevel := s.getTopLevelPackageElement(string(s.spdxDoc.SPDXIdentifier))
 	// adding top level package edge manually for all depends on package
-	if toplevel != nil {
-		preds.IsDependency = append(preds.IsDependency, common.CreateTopLevelIsDeps(toplevel[0], s.packagePackages, s.filePackages, "top-level package GUAC heuristic connecting to each file/package")...)
-		preds.HasSBOM = append(preds.HasSBOM, common.CreateTopLevelHasSBOM(toplevel[0], s.doc))
+	if topLevel != nil {
+		for _, topLevelPkg := range topLevel {
+			preds.HasSBOM = append(preds.HasSBOM, common.CreateTopLevelHasSBOM(topLevelPkg, s.doc))
+		}
+
+		if s.topLevelIsHeuristic {
```

Comment:
Will have to do something similar to CDX also. Currently we are just creating the `CreateTopLevelIsDeps` without any checks in the dependencies.

Reply:
https://github.com/guacsec/guac/issues/996